Contents/Resources/Series/Shoko Series Scanner.py

```diff
@@ -145,10 +147,50 @@ def Scan(path, files, mediaList, subdirs, language=None, root=None):
             except Exception as e:
                 Log.error("Error in Scan: '%s'" % e)
                 continue
-        
-        Log.info('stack media')
+
         Stack.Scan(path, files, mediaList, subdirs)
-        Log.debug('media list %s', mediaList)
+
+    if not path: # If current folder is root folder
+        Log.info("Manual call to group folders")
+        subfolders = subdirs[:]
+
+        while subfolders: # subfolder scanning queue
+            full_path = subfolders.pop(0)
+            path = os.path.relpath(full_path, root)
+
+            reverse_path = list(reversed(path.split(os.sep)))
+            
+            Log.info(''.ljust(100, '='))
+            Log.info('Started subfolder scan: %s', full_path)
+            Log.info(''.ljust(100, '='))
+
+            subdir_dirs, subdir_files = [], []
+
+            for file in os.listdir(full_path):
+                path_item = os.path.join(full_path, file) 
+                if os.path.isdir(path_item):
+                    subdir_dirs.append(path_item)
+                else:
+                    subdir_files.append(path_item)
+
+            Log.info("Sub-directories: %s", subdir_dirs)
+            Log.info("Files: %s", subdir_files)
+
+            for dir in subdir_dirs:
+                Log.info("[Added for scanning] " + dir) # Add the subfolder to subfolder scanning queue)
+                subfolders.append(dir)
+
+            grouping_dir = full_path.rsplit(os.sep, full_path.count(os.sep)-1-root.count(os.sep))[0]
+            if subdir_files and (len(reverse_path)>1 or subdir_dirs):
+                if grouping_dir in subdirs:
+                    subdirs.remove(grouping_dir)  #Prevent group folders from being called by Plex normal call to Scan()
+                Log.info("CALLING SCAN FOR FILES IN CURRENT FOLDER")
+                Scan(path, sorted(subdir_files), mediaList, [], language=language, root=root) 
```

Comment:
do the parameters have to be explicit here? since they are in a defined order above, and you keep that same naming convention `root=root` seems redundant, and seems more like a local variable assignment more than a param

Reply:
Again, copied code.. xD